snappi/snappicommon.py

```diff
@@ -417,12 +417,14 @@ def validate(self):
         for key, value in self._properties.items():
             self._validate_types(key, value)
     
-    def getproperty(self, name):
+    def getproperty(self, name, get_default=False):
         """ 
         getattr for snappi object
         """
         if self._properties.get(name) is not None:
             return self._properties[name]
+        elif get_default:
+            return getattr(self, name)
```

Comment:
as we discussed, this will mutate `self` which we're trying to avoid. if we wanted to mutate, we wouldn't need `getproperty()`